Make test_exception_difficulty hard as steel

Sometimes `TestKeysAuth.test_exception_difficulty()` fails on windows due to non-empty dir.
Two possible solutions are:

 1. Make sure all files are deleted in test cleanup/teardown
 1. Use `shutil.rmtree()`

```python
================================== FAILURES ===================================
___________________ TestKeysAuth.test_exception_difficulty ____________________
self = <tests.golem.core.test_keysauth.TestKeysAuth testMethod=test_exception_difficulty>
    def test_exception_difficulty(self):
        # given
        lower_difficulty = 0
        req_difficulty = 7
        priv_key = str(random())[2:]
        assert lower_difficulty < req_difficulty  # just in case
    
        keys_dir = KeysAuth._get_or_create_keys_dir(self.path)
        # create key that has difficulty lower than req_difficulty
        while True:
            ka = self._create_keysauth(lower_difficulty, priv_key)
            if not ka.is_difficult(req_difficulty):
                break
>           os.rmdir(keys_dir)  # to enable keys regeneration
E           OSError: [WinError 145] The directory is not empty: 'C:\\Users\\buildbot-worker\\AppData\\Local\\Temp\\golem-tests-r12xcnpx\\test_exception_difficultyh7zzwc7r\\keys'
tests\golem\core\test_keysauth.py:90: OSError
```

--- a/tests/golem/core/test_keysauth.py
+++ b/tests/golem/core/test_keysauth.py
@@ -1,4 +1,5 @@
 import os
+import shutil
 import time
 from random import random, randint
 from unittest.mock import patch
@@ -87,7 +88,7 @@ class TestKeysAuth(testutils.PEP8MixIn, testutils.TempDirFixture):
             ka = self._create_keysauth(lower_difficulty, priv_key)
             if not ka.is_difficult(req_difficulty):
                 break
-            os.rmdir(keys_dir)  # to enable keys regeneration
+            shutil.rmtree(keys_dir)  # to enable keys regeneration
 
         assert KeysAuth.get_difficulty(ka.key_id) >= lower_difficulty
         assert KeysAuth.get_difficulty(ka.key_id) < req_difficulty

